internal/statusapi/statusapi.go

```diff
@@ -0,0 +1,107 @@
+package statusapi
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/fabric8-services/fabric8-jenkins-proxy/internal/clients"
+	"github.com/fabric8-services/fabric8-jenkins-proxy/internal/configuration"
+	"github.com/fabric8-services/fabric8-jenkins-proxy/internal/proxy"
+	"github.com/julienschmidt/httprouter"
+	log "github.com/sirupsen/logrus"
+)
+
+//JenkinsStatusAPI contains API to check whether Jenkins for the current user is idle|running|starting
+type JenkinsStatusAPI interface {
+	Status(w http.ResponseWriter, r *http.Request, ps httprouter.Params)
+}
+
+// JenkinsStatusAPIImpl implements JenkinsStatusAPI
+type jenkinsStatusAPIImpl struct {
+	tenant *clients.Tenant
+	idler  clients.IdlerService
+}
+
+// JenkinsStatus represents whether jenkins is running|starting|idle
+type JenkinsStatus struct {
+	Status string `json:"status"`
+}
+
+// NewJenkinsStatusAPI creates a new instance of JenkinsStatusAPI
+func NewJenkinsStatusAPI(tenant *clients.Tenant, idler clients.IdlerService) JenkinsStatusAPI {
+	return &jenkinsStatusAPIImpl{
+		tenant: tenant,
+		idler:  idler,
+	}
+}
+
+// Status returns the Jenkins status for the current user
+func (api *jenkinsStatusAPIImpl) Status(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+
+	log.Infof("Status API ran")
+	tokenJSON, ok := r.URL.Query()["token_json"]
+	if !ok {
+		log.Errorf("Couldn't find token_json in the query")
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	namespace, err := processToken(tokenJSON[0], api.tenant)
+	if err != nil {
+		log.Error(err)
+		return
+	}
+	log.Infof("Found token info in the query. Namespace is %s and clusterURL is %s", namespace.Name, namespace.ClusterURL)
+
+	status, err := api.idler.Status(namespace.Name, namespace.ClusterURL)
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
+	if status == "idled" {
```

Comment:
instead of harded `"idled"` plesae consider using a constant like in https://github.com/sthaha/fabric8-jenkins-proxy/commit/97d7357d13e5543d9878d6018c4e09c9a90ee060

Reply:
and it should be `status != clients.Running`

Reply:
Done